git_machete/git_operations.py

```diff
@@ -15,40 +15,83 @@
 class AnyRevision(str):
     @staticmethod
     def of(value: str) -> Optional["AnyRevision"]:
-        return AnyBranch(value) if value else None
+        return AnyRevision(value) if value else None
+
+    def full_name(self) -> "AnyRevision":
+        return self
 
 
 class AnyBranch(AnyRevision):
     @staticmethod
     def of(value: str) -> Optional["AnyBranch"]:
         return AnyBranch(value) if value else None
 
+    def full_name(self) -> "AnyBranch":
+        return self
+
 
 class LocalBranch(AnyBranch):
     @staticmethod
     def of(value: str) -> Optional["LocalBranch"]:
         return LocalBranch(value) if value else None
 
+    def full_name(self) -> Optional["LocalBranchFullName"]:
+        return LocalBranchFullName.of(self)
+
+
+class LocalBranchFullName(LocalBranch):
+    @staticmethod
+    def of(value: str) -> Optional["LocalBranchFullName"]:
+        return LocalBranchFullName(f"refs/heads/{value}") if value else None
+
 
 class RemoteBranch(AnyBranch):
     @staticmethod
     def of(value: str) -> Optional["RemoteBranch"]:
         return RemoteBranch(value) if value else None
 
+    def full_name(self) -> Optional["RemoteBranchFullName"]:
+        return RemoteBranchFullName.of(self)
+
 
-class FullCommitSha(AnyRevision):
+class RemoteBranchFullName(RemoteBranch):
     @staticmethod
-    def of(value: str) -> Optional["FullCommitSha"]:
-        return FullCommitSha(value) if value else None
+    def of(value: str) -> Optional["RemoteBranchFullName"]:
+        return RemoteBranchFullName(f"refs/remotes/{value}") if value else None
```

Comment:
Hmmm this is confusing (since now `of` requires a short name, but returns a long one)... I'd rather make a distinction like:
```suggestion
    def of(value: str) -> Optional["RemoteBranchFullName"]:
        return RemoteBranchFullName(value) if value else None
    @staticmethod
    def from_short_name(value: RemoteBranchShortName) -> "RemoteBranchFullName":
        return self.of(f"refs/remotes/{value}")
```

Reply:
Similarly for the local branches